sources/packages/backend/libs/services/src/notifications/notification-message/notification-message.service.ts

```diff
@@ -13,26 +13,29 @@ export class NotificationMessageService extends RecordDataModelService<Notificat
   }
 
   /**
-   * Retrieves the template id of a notification message.
+   * Retrieves the notification message details of a notification message.
    * @param notificationMessageTypeId id of the user who will receive the message.
    * @param options options.
    * - `entityManager` external entity manager to run in a transaction.
-   * @returns template Id of the notification message.
+   * @returns notification details of the notification message.
    */
-  async getTemplateId(
+  async getNotificationMessageDetails(
     notificationMessageTypeId: NotificationMessageType,
     options?: { entityManager?: EntityManager },
-  ): Promise<string> {
+  ): Promise<Pick<NotificationMessage, "templateId" | "emailContacts">> {
     const notificationMessageRepo =
       options?.entityManager?.getRepository(NotificationMessage) ?? this.repo;
-    const notificationMessage = await notificationMessageRepo.findOne({
-      select: {
-        templateId: true,
+    const { templateId, emailContacts } = await notificationMessageRepo.findOne(
+      {
+        select: {
+          templateId: true,
+          emailContacts: true,
+        },
+        where: {
+          id: notificationMessageTypeId,
+        },
       },
-      where: {
-        id: notificationMessageTypeId,
-      },
-    });
-    return notificationMessage.templateId;
+    );
+    return { templateId, emailContacts };
```

Comment:
Instead of destructuring and creating another object literal you can just return it from the query:

```ts
return await notificationMessageRepo.findOne(
      {
        select: {
          templateId: true,
          emailContacts: true,
        },
        where: {
          id: notificationMessageTypeId,
        },
      },
    );

```

Reply:
I think await is not need also as you're returning a Promise.